src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/OxoGReadCounts.java

```diff
@@ -81,6 +84,155 @@ public void annotate(final ReferenceContext refContext,
+
+        // Initialize the mappings
+        final Map<Allele, MutableInt> f1r2Counts = variantAlleles.stream()
+                .collect(Collectors.toMap(a -> a, a -> new MutableInt(0)));
+
+        final Map<Allele, MutableInt> f2r1Counts = variantAlleles.stream()
+                .collect(Collectors.toMap(a -> a, a -> new MutableInt(0)));
+
+        final List<Allele> referenceAlleles = variantAlleles.stream().filter(a -> a.isReference() && !a.isSymbolic()).collect(Collectors.toList());
+        final List<Allele> altAlleles = variantAlleles.stream().filter(a -> a.isNonReference() && !a.isSymbolic()).collect(Collectors.toList());
+
+        if (referenceAlleles.size() != 1) {
+            logger.warn("Number of reference alleles does not equal  for VC: " + vc);
+        }
+
+        // We MUST have  exactly 1 non-symbolic reference allele, a read pileup,
+        if ((referenceAlleles.size() == 1) && (readPileup != null) && !referenceAlleles.get(0).isSymbolic()) {
+            final Allele referenceAllele = referenceAlleles.get(0);
+            Utils.stream(readPileup.iterator())
+                    .filter(r -> isUsableRead(r.getRead()))
+                    .forEach(r -> incrementCounts(r, f1r2Counts, f2r1Counts, referenceAllele, altAlleles, meanBaseQualityCutoff));
+        }
+
+        final int[] f1r2 = variantAlleles.stream().mapToInt(a -> f1r2Counts.get(a).intValue()).toArray();
+
+        final int[] f2r1 = variantAlleles.stream().mapToInt(a -> f2r1Counts.get(a).intValue()).toArray();
+
+        gb.attribute(GATKVCFConstants.F1R2_KEY, f1r2);
+        gb.attribute(GATKVCFConstants.F2R1_KEY, f2r1);
+    }
+
+    /**
+     *  If the allele is not in the count mappings, then it is not counted.  No exception will be thrown
+     *  Modifies count variables in place.
+     *
+     * @param pileupElement pileup overlapping the alleles
+     * @param f1r2Counts a mapping of allele to f1r2 counts
+     * @param f2r1Counts a mapping of allele to f2r1 counts
+     */
+    private static void incrementCounts(final PileupElement pileupElement, final Map<Allele, MutableInt> f1r2Counts,
+                                    final Map<Allele, MutableInt> f2r1Counts, final Allele referenceAllele,
+                                        final List<Allele> altAlleles, double meanBaseQualityCutoff) {
+        Map<Allele, MutableInt> countMap = f1r2Counts;
+        if (isF2R1(pileupElement.getRead())) {
+            countMap = f2r1Counts;
+        }
+
+        final boolean isRef = referenceAllele.basesMatch(getBasesForAlleleInRead(pileupElement, referenceAllele))
+                && !pileupElement.isBeforeDeletionStart() && !pileupElement.isBeforeInsertion();
+
+        Allele pileupAllele = null;
+        if (!isRef) {
+
+            for (Allele altAllele : altAlleles) {
+                final VariantContext.Type variantType = GATKProtectedVariantContextUtils.typeOfBiallelicVariant(referenceAllele, altAllele);
+
+                switch(variantType) {
```

Comment:
I don't think the `switch` is needed here.  This would read better to me as
```java
if (variantType == INDEL) {
   . . .
} else if (variantType == SNP || variantType == MNP) {
   . . .
}
```

Reply:
Done